### PR TITLE
log activation norms

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
Log statistics for residual stream and residual block updates

# Checklist
   - [ ] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [ ] Are all variables set with logical defaults that are backward compatible?
   - [ ] I attached link to neptune if it was necessary
   - [ ] Have you successfully executed the linter and tests?
   - [ ] Are all functions concise and don't require splitting?
   - [ ] Is there documentation for arguments and complex logic?
   - [ ] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [ ] Are all functions placed in appropriate files?
   - [ ] Is the PR name meaningful and descriptive?
   - [ ] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

**Comment by Simontwice** [comment]
[neptune](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=99f374ec-39cd-48c4-ac55-0f8f7c35500c&detailsTab=charts&shortId=LLMRANDOM-3028&type=run)

File: research/conditional/utils/layer_manager.py [hunk 1]
```diff
@@ -32,16 +32,31 @@ def __init__(
 
     def _register_layers(self, model):
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:33]: It assumes some block naming conventions. Do we have them written down somewhere?

File: research/conditional/utils/layer_manager.py [hunk 2]
```diff
@@ -32,16 +32,31 @@ def __init__(
...
+                    "feedforward"
+                    if hasattr(layer.layer, "feedforward")
+                    else "attention"
+                )
+                self._layers.append((f"{block_name}_{subblock_type}_residual", layer))
+                print(f"Registered layer {block_name}_{subblock_type}_residual")
+            elif "feedforward" in suffix:
+                block_name = self.extract_block_name(name)
                 self._layers.append((block_name, layer))
+                print(f"Registered layer {block_name}")
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:48]:
```suggestion
            block_name = self.extract_block_name(name)
            if "residual" in suffix:
                subblock_type = (
                    "feedforward"
                    if hasattr(layer.layer, "feedforward")
                    else "attention"
                )
                registered_name = f"{block_name}_{subblock_type}_residual"
            elif "feedforward" in suffix:
                registered_name = block_name
            self._layers.append((registered_name, layer))
            print(f"Registered layer {registered_name}")
```

**Review by crewtool — APPROVED** [review]
LGTM

File: research/conditional/tests/test_layer_manager.py [hunk 3]
```diff
@@ -71,29 +71,7 @@ class TestLearningStartAdatempPositive(GeneralTestCase):
...
-        model,
-        0,
-        0,
-        steps_until_start_temperature_learn=steps_until_start_temperature_learn,
-    )
-
-    for step in range(100):
-        layer_manager.manage_learnable_temperature(step)
-        for name, module in model.named_children():
-            if name.endswith("feedforward"):
```
Comment (maciek-pioro) [inline research/conditional/tests/test_layer_manager.py:96]: Was this section repeated?

File: research/conditional/utils/layer_manager.py (hunk 2 above)
Comment (maciek-pioro) [inline research/conditional/utils/layer_manager.py:48]: Spent a couple of minutes trying to understand that, but I lack background in our naming convention and logging. I am especially lost with `hasattr(layer.layer, "feedforward")`. 
@Simontwice can you either:
a) give a thorough explanation of this part of code
b) find a reviewer familiar with logging layers, e.g. @kuba-krj

**Review by maciek-pioro — CHANGES_REQUESTED** [review]
Left a couple of comments. Depending on what we do about the last comment, I will either re-review or just accept if another reviewer accepts.

**Comment by crewtool** [comment]
> Left a couple of comments. Depending on what we do about the last comment, I will either re-review or just accept if another reviewer accepts.

I already accepted.

**Comment by crewtool** [comment]
I totally agree with @maciek-pioro  (as I stated in my review comment)  we need to have a clear convention

File: research/conditional/tests/test_layer_manager.py (hunk 3 above)
Comment (Simontwice) [reply research/conditional/tests/test_layer_manager.py:96]: it was, somehow :D

File: research/conditional/utils/layer_manager.py (hunk 1 above)
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py:33]: i have not, i will attempt to put it down in a docstring

File: research/conditional/utils/layer_manager.py (hunk 2 above)
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py]: I think now it's much more self-explanatory, pls take a look @maciek-pioro

Comment (Simontwice) [reply research/conditional/utils/layer_manager.py]: applied

**Review by crewtool — APPROVED** [review]

**Comment by Simontwice** [comment]
@maciek-pioro please address

**Pull request merged**